### DISCUSSION TITLE
Minimum reaction time

### DISCUSSION THREAD

**Discussion by viviers** [jspsych/jsPsych#886] [Q&A] [answered]
Is there an easy way to not allow keyboard input until a certain reaction time is reached on a trial? My trials are produced by a `while` loop so I need to do it without disrupting the loop? I'm using `html-keyboard-response` for the majority of my trials.

I also need to set a minimum entry for the `survey-text` plugin too. They need to type 50 words but some participants are just typing 1 or 2 and going next.

**Comment by vijaymarupudi** [comment, marked as answer]
What I would recommend is having two trials for one actual study trial. The first trial would accept no keys and would just display the stimuli (see the `trial_duration` parameter). The second trial would accept keys and take the response.

**Reply by viviers** [reply]
The only problem with this is that the trial is in a `while` loop, so I have no idea to add another trial before every instance of the `while` loop. Thank you so much for the great idea though!

**Comment by jodeleeuw** [comment]
I think @vijaymarupudi's solution is the best way right now. We're tracking a possible implementation of minimum valid rt's here: #546

**Comment by jodeleeuw** [comment]
@viviers can you please separate the question about survey-text responses into a different discussion thread so that it is easier for folks to find the answer once it is answered?
